jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -109,6 +109,21 @@ private static native void checkpointJVMImpl(String imageDir,
 		});
 	}
 
+	/**
+	 * A hook can be one of the following states:
+	 * SINGLE_THREAD - a hook running when the single-threaded mode is enabled
+	 * MULTI_THREAD - a hook ruuning when the single-threaded mode is NOT enabled
+	 */
+	public static enum HookMode {
+		SINGLE_THREAD,
```

Comment:
Im thinking we should rename to, SINGLE_THREAD_MODE, and CONCURRENT_MODE to be more clear.

Reply:
Changed to `SINGLE_THREAD_MODE` and `CONCURRENT_MODE`.